From cb31b22122fdf1c59ce15d8bca7135da719331dd Mon Sep 17 00:00:00 2001
From: George Mocanu <george.mocanu@nxp.com>
Date: Wed, 22 Nov 2023 13:29:13 +0200
Subject: [PATCH] arm64: dts: s32g: GoldVIP-specific adaptations

Different updates on the the S32G dts for GoldVIP
usage:
  - update reserved memory ranges for the PFE
  buffers
  - add the SRAM memory range used for IPC
  - add MC_RGM and MC_ME nodes to provide support
  to their registers from user-space

Issue: GVIP-1193
Signed-off-by: George Mocanu <george.mocanu@nxp.com>
---
 arch/arm64/boot/dts/freescale/s32g.dtsi | 26 +++++++++++++++++++++----
 1 file changed, 22 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/s32g.dtsi b/arch/arm64/boot/dts/freescale/s32g.dtsi
index 7b20d20e51fc..8b0372ee3cee 100644
--- a/arch/arm64/boot/dts/freescale/s32g.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32g.dtsi
@@ -18,18 +18,26 @@
 
 / {
 	reserved-memory {
-		pfe_reserved_bmu2: pfebufs@34000000 {
+		ipcf_shm: sram@34500000 {
+			compatible = "nxp,s32g-ipcf-shm";
+			/* 1 MB */
+			reg = <0x0 0x34500000 0x0 0x100000>;
+			no-map;
+			status = "okay";
+		};
+
+		pfe_reserved_bmu2: pfebufs@34400000 {
 			compatible = "nxp,s32g-pfe-bmu2-pool";
 			/* BMU2: 512 KB */
-			reg = <0 0x34000000 0 0x80000>;
+			reg = <0 0x34400000 0 0x80000>;
 			no-map;
 			status = "okay";
 		};
 
-		pfe_reserved_rt: pfebufs@34080000 {
+		pfe_reserved_rt: pfebufs@34480000 {
 			compatible = "nxp,s32g-pfe-rt-pool";
 			/* RT: 128 KB */
-			reg = <0 0x34080000 0 0x20000>;
+			reg = <0 0x34480000 0 0x20000>;
 			no-map;
 			status = "okay";
 		};
@@ -51,6 +59,16 @@ pfe_reserved_bdr: pfebufs@835e0000 {
 		/* ends 0x83600000 */
 	};
 
+	mc_rgm: mc_rgm@40078000 {
+		compatible = "fsl,s32gen1-rgm", "nxp,s32g-rgm", "syscon";
+		reg = <0x0 0x40078000 0x0 0x3000>;
+	};
+
+	mc_me: mc_me@40088000 {
+		compatible = "fsl,s32gen1-mc_me", "nxp,s32g-mc_me", "syscon";
+		reg = <0x0 0x40088000 0x0 0x3000>;
+	};
+
 	soc {
 		siul2@4009c000 {
 			compatible = "simple-mfd";
-- 
2.25.1

